src/System.Text.Http/System/Text/Http/Parser/HttpParser.cs

```diff
@@ -107,271 +105,151 @@ public HttpParser(bool showErrorDetails)
                 span = PipelineExtensions.ToSpan(buffer.Slice(0, eol + s_Eol.Length));
             }
 
-            // Fix and parse the span
-            fixed (byte* data = &MemoryMarshal.GetReference(span))
-            {
-                ParseRequestLine(ref handler, data, span.Length);
-            }
+            // Parse the span
+            ParseRequestLine(ref handler, span);
 
             consumed = span.Length;
             return true;
         }
 
-        private unsafe void ParseRequestLine<T>(ref T handler, byte* data, int length) where T : IHttpRequestLineHandler
+        private unsafe void ParseRequestLine<T>(ref T handler, ReadOnlySpan<byte> data) where T : IHttpRequestLineHandler
```

Comment:
Similarly, here/elsewhere, if we are using span instead of ptr/length, we should remove the unsafe keyword.